### ISSUE TITLE
new pasta system test fails: podman networking with pasta(1) - Local forwarder, IPv4

### ISSUE THREAD

**Issue by Luap99** [containers/podman#17074]
The new VM images include pasta: https://github.com/containers/podman/pull/16525
One test is failing for unknown reasons, I can reproduce locally as well.
```
[+1142s] not ok 424 podman networking with pasta(1) - Local forwarder, IPv4
         # (from function `die' in file test/system/[helpers.bash, line 541](https://github.com/containers/podman/blob/193bac1b5deaf39ed4e04e903ddac09764dd4295/test/system/helpers.bash#L541),
         #  from function `run_podman' in file test/system/[helpers.bash, line 225](https://github.com/containers/podman/blob/193bac1b5deaf39ed4e04e903ddac09764dd4295/test/system/helpers.bash#L225),
         #  in test file test/system/[505-networking-pasta.bats, line 340](https://github.com/containers/podman/blob/193bac1b5deaf39ed4e04e903ddac09764dd4295/test/system/505-networking-pasta.bats#L340))
         #   `run_podman run --dns 198.51.100.1 \' failed
         # $ podman rm -t 0 --all --force --ignore
         # $ podman ps --all --external --format {{.ID}} {{.Names}}
         # $ podman images --all --format {{.Repository}}:{{.Tag}} {{.ID}}
         # quay.io/libpod/testimage:20221018 f5a99120db64
         # $ podman run --dns 198.51.100.1 --net=pasta:--dns-forward,198.51.100.1 quay.io/libpod/testimage:20221018 nslookup 127.0.0.1
         # ;; connection timed out; no servers could be reached
         # [ rc=1 (** EXPECTED 0 **) ]
         # #/vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
         # #| FAIL: exit code is 1; expected 0
         # #\^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
From https://api.cirrus-ci.com/v1/artifact/task/5686752053035008/html/sys-remote-fedora-37-rootless-host.log.html

I am going to skip this one for now so we can merge the image update and we can unskip it once this is fixed.

**Comment by rhatdan** [comment]
@Luap99 What is happening with this one?

**Comment by Luap99** [comment]
waiting on upstream to fix it
cc @sbrivio-rh

**Comment by sbrivio-rh** [comment]
Right, I forgot to update this ticket, thanks @Luap99 for the reminder.

I couldn't find any actual issue in pasta with this regard, I suspect there's an issue with how the test sources the address of the resolver.

Issue is on myself anyway, assigning.

**Comment by sbrivio-rh** [comment]
I can't assign it to myself, it seems. Feel free to do it if possible. Not that it really matters, this is on my list anyway.

**Comment by sbrivio-rh** [comment]
I think this is now fixed in `passt-0^20230227.gc538ee8-1.fc37`, not yet available in the "testing-updates" repositories because I'm still waiting for the previous version to reach "stable".

The issue should be fixed by:
`   `  https://passt.top/passt/commit/?id=bad252687271c0255f6a077f19cbc19aa0427f8d

we had captures available from this report, which made debugging quite straightforward.

I'll submit a merge request and ask to update CI images once I can finally get the updated version in Fedora 37 repositories.

**Comment by sbrivio-rh** [comment]
@cevich This issue should be fixed in `passt-0^20230227.gc538ee8-1.fc37` and I could submit a merge request (re-enabling the test) to try it out. That package version is still in the "testing" repository though, so I wonder: would it take less effort for you if we just used "stable" packages (and waited a bit here... but also in the general case)?

**Comment by cevich** [comment]
The point of CI is to try and catch problems early. But you're insight is also accurate, in that we don't want CI to be a PITA of constant updates.  In the past for "new" technologies with a lot of churn, we've simply setup CI to grab & use the latest/greatest upstream builds.  Bypassing distro packaging entirely.  Then later, once the churn settled down, we adjusted accordingly.

That said, I'm not bothered by building new images every other week or so.  More than that and it starts to feel like a hassle.  I'm kind of on the fence with Pasta, it seems (IMHO) stable enough, without "too much" churn that the distro-packages are okay for CI.  Do you feel differently?

**Comment by sbrivio-rh** [comment]
> That said, I'm not bothered by building new images every other week or so. More than that and it starts to feel like a hassle.

Okay, thanks, that's what I needed to know. :)

> I'm kind of on the fence with Pasta, it seems (IMHO) stable enough, without "too much" churn that the distro-packages are okay for CI. Do you feel differently?

No no, I also think so.

Besides, I maintain both Debian and Fedora packages, so I'm updating them whenever there's an issue that looks critical enough (including anything that might make the related Podman tests fail).

I just wanted to get an understanding of how frequently I can reasonably ask you to update images, because I have no clue about the effort involved.

...so, yeah, if you could update the Fedora image to use `passt-0^20230227.gc538ee8-1.fc37` (https://bodhi.fedoraproject.org/updates/FEDORA-2023-ae66901b6a) and the Debian image to use `passt (0.0~git20230227.c538ee8-1)` (it's already in unstable) at some point this week, that would be great, so I can hopefully re-enable this CI test. Thanks!

**Comment by cevich** [comment]
> I have no clue about the effort involved.

Neither do I lol.  Most of the time it's just a matter of opening a PR and waiting.  But every now and again a monkey-wrench or blob of duct-tape falls into the gears.  Let's hope it's the former and not the later...

**Comment by cevich** [comment]
...yay, @sbrivio-rh the build went smoothly.  Use `c20230307t192532z-f37f36d12` as the `IMAGE_SUFFIX` in `.cirrus.yml` for `passt-0^20230227.gc538ee8-1.fc37`.

**Comment by sbrivio-rh** [comment]
@cevich great, thanks a lot! I'll try in a bit.
